ArmPlatformPkg/Drivers/PL061GpioDxe/PL061Gpio.c

```diff
@@ -100,7 +100,7 @@ PL061SetPins (
   IN UINTN  Value
   )
 {
-  MmioWrite8 (PL061EffectiveAddress (Address, Mask), Value);
+  MmioWrite8 (PL061EffectiveAddress (Address, Mask), (UINT8)Value);

```

Comment:
Instead of a cast, wouldn't it be better to change the type of the "Value" parameters to UINT8? After all the GPIO spec [1]  specifies the register width to be 8-bits.

(And while we're at it, we should probably look at the return type of PL061GetPins and change the Mask values in PL061GetPins and PL061EffectiveAddress to UINT8?)

Reply:
Agree should change the "Value" parameter to UINT8.

However, PL061GetPins and PL061EffectiveAddres return type is UINTN. So i don't think the Mask should be changed to UINT8. 

`STATIC
UINTN
EFIAPI
PL061EffectiveAddress (
  IN UINTN  Address,
  IN UINTN  Mask
  )
{
  return ((Address + PL061_GPIO_DATA_REG_OFFSET) + (Mask << 2));
}`
Shows the Address getting added to the Mask. So if I changed the type of Mask to UINT8 here, i would have to typecast it back to UINTN.

Reply:
I see that all calls to PL061GetPins and PL061SetPins use GPIO_PIN_MASK - which now returns an 8-bit value - to pass the Mask. Since the Mask is expected to be 8-bit, I guess it's reasonable to use UINT8 in the parameters and only cast it to UINTN inside PL061EffectiveAddress.

PL061GetPins returns the result of an MmioRead8() call which returns an 8-bit value so I think it makes more sense to change the return type to UINT8 to reflect that

Reply:
Done. Thanks @mgamal